aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -365,7 +360,9 @@ def get_record(self, event: Dict[str, Any]) -> DataRecord:
 
         record = self._get_record(idempotency_key)
 
-        if self.use_local_cache:
+        # We can't cache "INPROGRESS" records as we have no way to reflect updates that can happen outside of the
+        # execution environment
+        if self.use_local_cache and not record.status == STATUS_CONSTANTS["INPROGRESS"]:
```

Comment:
@cakepietoast  alternatively you could just modify `_save_to_cache` to implement this check. 
Easier to test and guarantee we never save "INPROGRESS" records

Reply:
Updates to _retrieve_from_cache:

```python3
    def _retrieve_from_cache(self, idempotency_key: str):
        cached_record = self._cache.get(idempotency_key)
        if cached_record:
            if not cached_record.is_expired:
                return cached_record
            logger.debug(f"Removing expired local cache record for idempotency key: {idempotency_key}")
            self._delete_from_cache(idempotency_key)
```

Update `_save_to_cache` would be 

```python3
    def _save_to_cache(self, data_record: DataRecord):
        # We can't cache "INPROGRESS" records as we have no way to reflect updates that can happen outside of the
        # execution environment
        if data_record.status == STATUS_CONSTANTS["INPROGRESS"]:
            return

        self._cache[data_record.idempotency_key] = data_record
```

Modify existing test `test_idempotent_lambda_in_progress_with_cache` to do:
```python3
save_to_cache_spy.assert_called()
```

Add new test for `_save_to_cache`

```python3
@pytest.mark.parametrize("persistence_store", [{"use_local_cache": True}], indirect=True)
def test_in_progress_never_saved_to_cache(persistence_store):
    # GIVEN a data record with status "INPROGRESS"
    # and persistence_store has use_local_cache = True
    data_record = DataRecord("key", status="INPROGRESS")

    # WHEN saving to local cache
    persistence_store._save_to_cache(data_record)

    # THEN don't save to local cache
    assert persistence_store._cache.get("key") is None
```